### DISCUSSION TITLE
How to exchange data with outside of the Behavior Tree？

### DISCUSSION THREAD

**Discussion by Senkuu-ai** [BehaviorTree/BehaviorTree.CPP#783] [Q&A]

**Comment by IversonHuang** [comment]
It can be used Blackboard::set() to set data to your tree, and Blackboard::get() to get data inside tree. But this might only get the data of the main tree, the data of subtree cannot be reached, useless you set __shared_blackboard for your subtree or remapping the data of subtree to maintree.
